### PR TITLE
[CPDLP-2879] Add missing `ecf_id` to statements

### PR DESCRIPTION
**Pull request by leandroalemao**
### Context

Ticket: [CPDLP-2879](https://dfedigital.atlassian.net/browse/CPDLP-2879)

We need to maintain all existing ecf ids in the API. This also applies to statements.

### Changes proposed in this pull request

- Add `ecf_id` to the statements table
- Migrate the `ecf_id` when copying over statement data
- Surface `ecf_id` and the id of the statement in the API
- Add to seeds as well as a generated UUID

[CPDLP-2879]: https://dfedigital.atlassian.net/browse/CPDLP-2879?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: db/migrate/20240312212008_add_ecf_id_to_statements.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+class AddEcfIdToStatements < ActiveRecord::Migration[7.1]
+  def change
+    add_column :statements, :ecf_id, :text, null: true
+    add_index :statements, :ecf_id, unique: true
```
Comment (cwrw) [inline db/migrate/20240312212008_add_ecf_id_to_statements.rb:6]: could we add the uniqueness validation to the model as well?

File: app/serializers/api/v3/statement_serializer.rb [hunk 2]
```diff
@@ -17,6 +17,7 @@ class AttributesSerializer < Blueprinter::Base
         field :deadline_date, name: :cut_off_date, datetime_format: "%Y-%m-%d"
         field :payment_date, datetime_format: "%Y-%m-%d"
         field(:paid) { |s, _| s.payment_date.present? }
+        field :ecf_id
```
Comment (cwrw) [inline app/serializers/api/v3/statement_serializer.rb:20]: could we use this instead of the actual `id` and not a new field?

File: db/ecf_schema.rb [hunk 3]
```diff
@@ -207,6 +207,7 @@
     t.datetime "updated_at", null: false
     t.integer "disable_from_year"
     t.boolean "listed", default: false, null: false
+    t.integer "listed_for_school_type_codes", default: [], array: true
```
Comment (mooktakim) [inline db/ecf_schema.rb:210]: I'm not sure if these changes are intentional; seems unrelated to the PR.

**Review by mooktakim — APPROVED** [review]
Minor comment about `ecf_schema.rb`, otherwise looks good

File: db/ecf_schema.rb (hunk 3 above)
Comment (cwrw) [reply db/ecf_schema.rb:210]: I think its ok to keep this this time as we have to keep the ecf schema in sync with ECF

**Review by cwrw — APPROVED** [review]
